OBSDOCS-944 - Power Monitoring - Add Redfish support in power monitoring (#76231)
openshift/openshift-docs: modules/power-monitoring-configuring-kepler-redfish.adoc

      skipSSLVerify: false <3>
----
<1> Specifies the name of the secret that contains the credentials for accessing the Redfish server. This field is required.
<2> Controls the frequency at which the power information is queried from Redfish. This field is optional. The default value is `60s`.
<3> Controls if {PM-kepler} skips verifying the Redfish server certificate. This field is optional. The default value is `false`.
+
[NOTE]
====
The `openshift-power-monitoring` namespace is created after the {PM-kepler} is deployed.
====
. Create the `redfish.csv` file with this data format `your_kubelet_node_name, redfish_username, redfish_password, https://redfish_ip_or_hostname/`.
+
.Example output
[source,csv]
----
control-plane,dummyuser,dummypass,https://redfish.nodes.example.com
worker-1,dummyuser,dummypass,https://redfish.nodes.example.com
worker-2,dummyuser,dummypass,https://another.redfish.nodes.example.com
----
. Create the secret under `openshift-power-monitoring` namespace. You must create the secret with the following conditions:
* The secret type is `Opaque`
* The credentials are stored under the `redfish.csv` key in the `data` field of the secret.
+
[source,terminal]
----

> $ oc -n openshift-power-monitoring \
>   create secret generic redfish-secret \
>   --from-file=redfish.csv

Reviewer: ```suggestion
$ oc -n openshift-power-monitoring \
      create secret generic redfish-secret \
      --from-file=redfish.csv
```

----
+
.Example output
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: redfish-scret
data:
  redfish.csv: YmFyCg==
----
[IMPORTANT]
====
The {PM-kepler} deployment will not continue until the Redfish secret is created. You can find this information in the status of a {PM-kepler} instance.

> ====

Reviewer: The rendering seems off. Are we missing a `+` ?

Author: Yes. fixing this. Thank you!

// Module included in the following assemblies:

// * power_monitoring/configuring-power-monitoring.adoc

:_mod-docs-content-type: PROCEDURE
[id="power-monitoring-configuring-kepler-redfish_{context}"]
= Configuring {PM-kepler} to use Redfish

> You can configure the {PM-kepler} to use the Redfish as the source for platform power.

Author: Kepler is responsible for monitoring the power usage of containers running. I think "power" is the correct usage here.

> You can configure the {PM-kepler} to use the Redfish as the source for platform power.

Reviewer 2: Maybe?

You can configure {PM-kepler} to use Redfish as the source for running/hosting containers. {PM-kepler} can then monitor the power usage of these containers

.Prerequisites
* You have access to the {product-title} web console.
* You are logged in as a user with the `cluster-admin` role.
* You have installed the {PM-operator}.

.Procedure

. In the *Administrator* perspective of the web console, go to *Operators* -> *Installed Operators*.

. Click *{PM-title-c}* from the *Installed Operators* list and go to the *{PM-kepler}* tab.

. Click *Create {PM-kepler}*. If you already have a {PM-kepler} instance created, click *Edit Kepler*.

> . Configure {PM-kepler} instance’s `.spec.exporter.redfish` specifying the mandatory `secretRef` and optional `probeInterval` and `skipSSLVerify`.

Reviewer 2: ```suggestion
. Configure `.spec.exporter.redfish` of the {PM-kepler} instance by specifying the mandatory `secretRef` parameter. You can also configure the optional `probeInterval` and `skipSSLVerify` parameters to meet your needs.
```

The customer should decide if they want to configure anything optional. The step seems to indicate that the parameters/fields are needed.

Author: I really like your suggestion in here. Is it okay if I use "field" instead of parameters?

Reviewer 2: For sure. Thanks!

+
.Example output
[source,yaml]
----
apiVersion: kepler.system.sustainable.computing.io/v1alpha1
kind: Kepler
metadata:
  name: kepler
spec:
  exporter:

>     deployment:  ... 

Reviewer 2: I'm not sure what the three dots mean here. The "Example Kepler resource with default configuration" uses a different approach. How about:

deployment:
# ...

Author: So, under "deployment", you got 3 more fields which isn't expanded here. The engineering team only wants to show the fields under "redfish". FYI, both are optional tasks and not mandatory for users to perform and Kepler would still be created without specifying "deployment" or "redfish".

    redfish:
      secretRef: <name of the secret> required <1>
      probeInterval: 60s <2>

>       skipSSLVerify: false <3>

Reviewer 2: ```suggestion
      skipSSLVerify: false <3>
# ...
```

----

> <1> Specifies the name of the secret that contains the credentials for accessing the Redfish server. This field is required.

Reviewer 2: I think frontloading the callouts with "Required/no required" would help customers instantly know what is mandatory before they read a description.

Author: Rephrased this to:
<1> Required: Specifies the name of the secret that contains the credentials for accessing the Redfish server.
<2> Optional: Controls the frequency at which the power information is queried from Redfish. The default value is `60s`.
<3> Optional: Controls if {PM-kepler} skips verifying the Redfish server certificate. The default value is `false`.

<2> Controls the frequency at which the power information is queried from Redfish. This field is optional. The default value is `60s`.
<3> Controls if {PM-kepler} skips verifying the Redfish server certificate. This field is optional. The default value is `false`.
+
[NOTE]
====

> The `openshift-power-monitoring` namespace is created after the {PM-kepler} is deployed.

Reviewer 2: ```suggestion
After the {PM-kepler} is deployed, the  `openshift-power-monitoring` namespace is created.
```

====

> . Create the `redfish.csv` file with this data format: `your_kubelet_node_name, redfish_username, redfish_password, https://redfish_ip_or_hostname/`.

Reviewer 2: ```suggestion
. Create the `redfish.csv` file with the following data format: `<your_kubelet_node_name>, <redfish_username>, <redfish_password>, https://<redfish_ip_or_hostname>/`.
```

+
.Example output
[source,csv]
----
control-plane,dummyuser,dummypass,https://redfish.nodes.example.com
worker-1,dummyuser,dummypass,https://redfish.nodes.example.com
worker-2,dummyuser,dummypass,https://another.redfish.nodes.example.com
----

> . Create the secret under `openshift-power-monitoring` namespace. You must create the secret with the following conditions:

Reviewer 2: ```suggestion
. Create the secret under the `openshift-power-monitoring` namespace. You must create the secret with the following conditions:
```

> * The secret type is `Opaque`

Reviewer 2: ```suggestion
* The secret type is `Opaque`.
```

* The credentials are stored under the `redfish.csv` key in the `data` field of the secret.

> +

Reviewer 2: Example heading would be useful.

[source,terminal]
----
$ oc -n openshift-power-monitoring \
      create secret generic redfish-secret \
      --from-file=redfish.csv
----
+
.Example output
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: redfish-scret
data:

>   redfish.csv: YmFyCg==

Reviewer 2: ```suggestion
  redfish.csv: YmFyCg==
  # ...
```

// Module included in the following assemblies:

// * power_monitoring/configuring-power-monitoring.adoc

:_mod-docs-content-type: PROCEDURE
[id="power-monitoring-configuring-kepler-redfish_{context}"]
= Configuring {PM-kepler} to use Redfish

> You can configure {PM-kepler} to use Redfish as the source for running/hosting containers. {PM-kepler} can then monitor the power usage of these containers.

Reviewer 2: ```suggestion
You can configure {PM-kepler} to use Redfish as the source for running or hosting containers. {PM-kepler} can then monitor the power usage of these containers.
```

.Procedure

. In the *Administrator* perspective of the web console, go to *Operators* -> *Installed Operators*.

. Click *{PM-title-c}* from the *Installed Operators* list and go to the *{PM-kepler}* tab.

. Click *Create {PM-kepler}*. If you already have a {PM-kepler} instance created, click *Edit Kepler*.

. Configure `.spec.exporter.redfish` of the {PM-kepler} instance by specifying the mandatory `secretRef` field. You can also configure the optional `probeInterval` and `skipSSLVerify` fields to meet your needs.
+
.Example output
[source,yaml]
----
apiVersion: kepler.system.sustainable.computing.io/v1alpha1
kind: Kepler
metadata:
  name: kepler
spec:
  exporter:
    deployment:
# ... 
    redfish:
      secretRef: <name of the secret> required <1>
      probeInterval: 60s <2>
      skipSSLVerify: false <3>
# ...
----
<1> Required: Specifies the name of the secret that contains the credentials for accessing the Redfish server.
<2> Optional: Controls the frequency at which the power information is queried from Redfish. The default value is `60s`.
<3> Optional: Controls if {PM-kepler} skips verifying the Redfish server certificate. The default value is `false`.
+
[NOTE]
====

> After the {PM-kepler} is deployed, the `openshift-power-monitoring` namespace is created.

Reviewer 2: ```suggestion
After {PM-kepler} is deployed, the `openshift-power-monitoring` namespace is created.
```

====
. Create the `redfish.csv` file with the following data format: `<your_kubelet_node_name>, <redfish_username>, <redfish_password>, https://<redfish_ip_or_hostname>/`.
+

> .Example output

Reviewer 3: Same re: not really being "output":
```suggestion
.Example `redfish.csv` file
```

[source,csv]
----

> control-plane,dummyuser,dummypass,https://redfish.nodes.example.com
> worker-1,dummyuser,dummypass,https://redfish.nodes.example.com
> worker-2,dummyuser,dummypass,https://another.redfish.nodes.example.com

Reviewer 3: There isn't any explicit guidance on the word "dummy" in the ISG or SSG or any of our guides I could find, but could these instances be just "example" instead? Or is there a technical reason it should be "dummy"?
```suggestion
control-plane,exampleuser,examplepass,https://redfish.nodes.example.com
worker-1,exampleuser,examplepass,https://redfish.nodes.example.com
worker-2,exampleuser,examplepass,https://another.redfish.nodes.example.com
```
Just feels similar to the reasoning behind not using "foo"; also a little concerned about potential translation issues / unintended meaning.

----

> . Create the secret under the `openshift-power-monitoring` namespace. You must create the secret with the following conditions:
> * The secret type is `Opaque`.
> * The credentials are stored under the `redfish.csv` key in the `data` field of the secret.
> +

Reviewer 3: This unordered list in the middle of the command is making the indentation get off for the command block that follows. You can use an [open block](https://docs.asciidoctor.org/asciidoc/latest/blocks/open-blocks) (`--`) to fix:
```suggestion
. Create the secret under the `openshift-power-monitoring` namespace. You must create the secret with the following conditions:
+
--
* The secret type is `Opaque`.
* The credentials are stored under the `redfish.csv` key in the `data` field of the secret.
--
+
```

> .Example

Reviewer 3: Don't need a block title for the command, only the example output (per [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output)).
```suggestion
```

[source,terminal]
----

> $ oc -n openshift-power-monitoring \
>       create secret generic redfish-secret \
>       --from-file=redfish.csv

Reviewer 3: Just a minor suggestion, lining the spacing up a little differently. 6 spaces seemed like a lot; I believe 4 might be the most common around the repo:
```suggestion
$ oc -n openshift-power-monitoring \
    create secret generic redfish-secret \
    --from-file=redfish.csv
```
or maybe just 2 to line it up with the first command after the prompt:
```suggestion
$ oc -n openshift-power-monitoring \
    create secret generic redfish-secret \
    --from-file=redfish.csv
```

Reviewer: @adellape , @shreyasiddhartha , my preference would the first version (to **not** lineup) to indicate that line 2 and 3 are continuation of the first line; i.e. its a single command.

:_mod-docs-content-type: PROCEDURE
[id="power-monitoring-configuring-kepler-redfish_{context}"]
= Configuring {PM-kepler} to use Redfish

You can configure {PM-kepler} to use Redfish as the source for running or hosting containers. {PM-kepler} can then monitor the power usage of these containers.

.Prerequisites
* You have access to the {product-title} web console.
* You are logged in as a user with the `cluster-admin` role.
* You have installed the {PM-operator}.

.Procedure

. In the *Administrator* perspective of the web console, click *Operators* -> *Installed Operators*.

. Click *{PM-title-c}* from the *Installed Operators* list and click the *{PM-kepler}* tab.

. Click *Create {PM-kepler}*. If you already have a {PM-kepler} instance created, click *Edit Kepler*.

. Configure `.spec.exporter.redfish` of the {PM-kepler} instance by specifying the mandatory `secretRef` field. You can also configure the optional `probeInterval` and `skipSSLVerify` fields to meet your needs.
+
.Example {PM-kepler} instance
[source,yaml]
----
apiVersion: kepler.system.sustainable.computing.io/v1alpha1
kind: Kepler
metadata:
  name: kepler
spec:
  exporter:
    deployment:
# ... 
    redfish:

>       secretRef: <secret name> required <1>

Reviewer 3: Still missing the underscore here:
```suggestion
      secretRef: <secret_name> required <1>
```

      probeInterval: 60s <2>
      skipSSLVerify: false <3>
# ...
----
<1> Required: Specifies the name of the secret that contains the credentials for accessing the Redfish server.
<2> Optional: Controls the frequency at which the power information is queried from Redfish. The default value is `60s`.
<3> Optional: Controls if {PM-kepler} skips verifying the Redfish server certificate. The default value is `false`.
+
[NOTE]
====
After {PM-kepler} is deployed, the `openshift-power-monitoring` namespace is created.
====
. Create the `redfish.csv` file with the following data format:
+
[source,csv]
----

> <your_kubelet_node_name>,<redfish_username>,<redfish_password>,\https://<redfish_ip_or_hostname>/

Reviewer 3: Sorry, I accidentally left the `\` solution in when I was copy-pasting around in my original comment. The `\` is only necessary when it's in monospace/backticks in-line of a sentence, but not when it's a full codeblock like this:
```suggestion
<your_kubelet_node_name>,<redfish_username>,<redfish_password>,https://<redfish_ip_or_hostname>/
```
